pkg/operator/controller/ingress_status.go

```diff
@@ -205,3 +209,80 @@ func getEventsByReason(events []corev1.Event, component, reason string) []corev1
 	}
 	return filtered
 }
+
+func computeDNSStatus(ic *operatorv1.IngressController, wildcardRecord *iov1.DNSRecord, dnsConfig *configv1.DNS) []operatorv1.OperatorCondition {
+	if dnsConfig.Spec.PublicZone == nil && dnsConfig.Spec.PrivateZone == nil {
+		return []operatorv1.OperatorCondition{
+			{
+				Type:    operatorv1.DNSManagedIngressConditionType,
+				Status:  operatorv1.ConditionFalse,
+				Reason:  "NoDNSZones",
+				Message: "No DNS zones are defined in the cluster dns config.",
+			},
+		}
+	}
+
+	if ic.Status.EndpointPublishingStrategy.Type != operatorv1.LoadBalancerServiceStrategyType {
+		return []operatorv1.OperatorCondition{
+			{
+				Type:    operatorv1.DNSManagedIngressConditionType,
+				Status:  operatorv1.ConditionFalse,
+				Reason:  "UnsupportedEndpointPublishingStrategy",
+				Message: "The endpoint publishing strategy doesn't support DNS management.",
+			},
+		}
+	}
+
+	conditions := []operatorv1.OperatorCondition{
+		{
+			Type:    operatorv1.DNSManagedIngressConditionType,
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "Normal",
+			Message: "DNS management is supported and zones are specified in the cluster DNS config.",
+		},
+	}
+
+	switch {
+	case wildcardRecord == nil:
+		conditions = append(conditions, operatorv1.OperatorCondition{
+			Type:    operatorv1.DNSReadyIngressConditionType,
+			Status:  operatorv1.ConditionFalse,
+			Reason:  "RecordNotFound",
+			Message: "The wildcard record resource was not found.",
+		})
+	case len(wildcardRecord.Status.Zones) == 0:
+		conditions = append(conditions, operatorv1.OperatorCondition{
+			Type:    operatorv1.DNSReadyIngressConditionType,
+			Status:  operatorv1.ConditionFalse,
+			Reason:  "NoZones",
+			Message: "The record isn't present in any zones.",
+		})
+	case len(wildcardRecord.Status.Zones) > 0:
+		failedZones := []configv1.DNSZone{}
+		for _, zone := range wildcardRecord.Status.Zones {
+			for _, cond := range zone.Conditions {
+				if cond.Type == iov1.DNSRecordFailedConditionType {
```

Comment:
Need the following: ` && cond.Status == string(operatorv1.ConditionTrue)`.

Reply:
Fixed